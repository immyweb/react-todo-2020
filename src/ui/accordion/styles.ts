import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
