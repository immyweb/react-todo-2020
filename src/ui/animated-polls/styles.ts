import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { fonts } from '../../styles/fonts';
import { H2, P } from '../typography/typography';

// Typography

export const APHeadline = glamorous(H2)({
  label: 'AP-headline',
  fontSize: 42,
  textTransform: 'uppercase',
  textAlign: 'center',
  color: colours.$white,
  [mediaQueries.tabletPortrait]: {
    fontSize: 60,
    lineHeight: 1,
  },
});

export const APCopy = glamorous(P)({
  label: 'AP-copy',
  fontSize: 16,
  textAlign: 'center',
  color: colours.$white,
  lineHeight: 1.13,
  marginBottom: 25,
  [mediaQueries.tabletPortrait]: {
    fontSize: 24,
  },
});

// Animated Polls

export const APoll = glamorous.section({
  label: 'AP',
  position: 'relative',
});

// Modal

export const ModalContainer = glamorous.div<{ open: boolean }>(
  {
    label: 'modal-container',
    position: 'fixed',
    width: '100%',
    height: '100%',
    top: 0,
    left: 0,
    zIndex: 160,
    [mediaQueries.tabletPortrait]: {
      position: 'absolute',
      height: 900,
      zIndex: 40,
    },
  },
  ({ open }) => ({
    visibility: open ? 'visible' : 'hidden',
  }),
);

export const ModalInner = glamorous.div({
  label: 'modal-inner',
  position: 'relative',
  height: '100vh',
  width: '100%',
  top: 0,
  background: colours.$black,
  overflowY: 'hidden',
  msOverflowStyle: '-ms-autohiding-scrollbar',
  [mediaQueries.tabletPortrait]: {
    height: 900,
  },
});

export const ModalContent = glamorous.div({
  label: 'modal-content',
  position: 'relative',
  overflowY: 'auto',
  msOverflowStyle: '-ms-autohiding-scrollbar',
  height: '100%',
});

export const ModalHeadline = glamorous.h2({
  label: 'modal-headline',
  textTransform: 'uppercase',
  textAlign: 'center',
  fontSize: 24,
  fontFamily: fonts.$theSunHeavyCondensed,
  marginBottom: 20,
  lineHeight: 1.1,
  padding: '0 25px',
  [mediaQueries.tabletPortrait]: {
    padding: 0,
  },
});

export const BtnClose = glamorous.div({
  label: 'modal-btn-close',
  textAlign: 'center',
  position: 'absolute',
  top: 23,
  right: 16,
  height: 28,
  background: 'none',
  borderRadius: '50%',
  ':hover': {
    cursor: 'pointer',
  },
  [mediaQueries.tabletPortrait]: {
    top: 35,
    right: 35,
  },
});

// Teaser screen

export const APTeaser = glamorous.section({
  label: 'AP-teaser',
  width: '100%',
  position: 'relative',
  minHeight: 520,
  [mediaQueries.tabletPortrait]: {
    height: 900,
    overflow: 'hidden',
  },
});

export const APTeaserLogo = glamorous.img<{ backgroundColour: string }>(
  {
    label: 'AP-teaser-logo',
    position: 'absolute',
    zIndex: 3,
    top: 8,
    left: 8,
    width: 72,
    padding: '8px 4px',
    [mediaQueries.mobileInner]: {
      top: 20,
      left: 20,
    },
  },
  ({ backgroundColour }) => ({
    backgroundColor: backgroundColour,
  }),
);

export const APTeaserImage = glamorous.div({
  label: 'AP-teaser-image',
  width: '100%',
  position: 'absolute',
  bottom: 0,
  left: 0,
  height: '100%',
  backgroundBlendMode: 'multiply',
  backgroundSize: 'cover',
  backgroundPosition: 'top center',
  backgroundRepeat: 'no-repeat',
  backgroundImage: `linear-gradient(to bottom, ${
    colours.$black
  } 0%, transparent 20%, transparent 50%, ${colours.$black} 81%, ${colours.$black} 100%)`,
  [mediaQueries.mobileInner]: {
    backgroundImage: `linear-gradient(to bottom, ${
      colours.$black
    } 0%, transparent 20%, transparent 50%, ${colours.$black} 81%, ${colours.$black} 100%)`,
  },
});

export const APTeaserVideo = glamorous.video({
  label: 'AP-teaser-video',
  width: '100%',
});

export const APTeaserInner = glamorous.div({
  label: 'AP-teaser-inner',
  position: 'absolute',
  bottom: 20,
  margin: '0 40px',
  [mediaQueries.mobileInner]: {
    bottom: 40,
    margin: '0 100px',
  },
});

// Intro Screen

export const APIntro = glamorous.section({
  label: 'AP-intro',
  width: '100%',
  position: 'relative',
  height: '100vh',
  background: colours.$black,
  [mediaQueries.mobileInner]: {
    height: 900,
    background: 'none',
  },
});

export const APIntroImg = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'AP-intro-img',
    position: 'absolute',
    width: '100%',
    top: 0,
    left: 0,
    height: 520,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
      height: 900,
    },
  }),
);

export const APIntroCopy = glamorous.div({
  label: 'AP-intro-copy',
  position: 'absolute',
  width: '100%',
  bottom: 0,
  padding: '0 40px 30px',
  background: colours.$black,
  [mediaQueries.mobileInner]: {
    paddingBottom: 15,
    background: 'none',
  },
});

export const APIntroLink = glamorous.a({
  label: 'AP-intro-link',
  color: colours.$midGrey,
  textDecoration: 'none',
});

export const APIntroHashTag = glamorous.span({
  label: 'AP-intro-hashtag',
  color: colours.$white,
  letterSpacing: 1.4,
});

// Card Screen

export const APCard = glamorous(APIntro)({
  label: 'AP-card',
});

export const APCardImg = glamorous(APIntroImg)({
  label: 'AP-card-image',
  height: '100vh',
  [mediaQueries.mobileInner]: {
    height: 900,
  },
});

export const APCardCopy = glamorous(APIntroCopy)({
  label: 'AP-card-copy',
  background: colours.$black,
  paddingTop: 15,
  minHeight: 165,
  ':before': {
    content: ' ',
    position: 'absolute',
    width: '100%',
    left: 0,
    top: -100,
    height: 100,
    backgroundImage: `linear-gradient(to bottom, rgba(0, 0, 0, 0), ${colours.$black})`,
  },
  [mediaQueries.mobileInner]: {
    background: colours.$black,
    minHeight: 'initial',
  },
});

// Vote Screen

export const APVote = glamorous(APIntro)({
  label: 'AP-vote',
  marginBottom: 0,
});

export const APVoteImgBkgnd = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'AP-vote-img-bkgnd',
    position: 'absolute',
    width: '100%',
    top: 0,
    left: 0,
    height: '100vh',
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const APVoteImgBkgndCover = glamorous.div({
  label: 'AP-vote-img-bkgnd-cover',
  position: 'absolute',
  top: 0,
  left: 0,
  width: '100%',
  height: '100vh',
  background: colours.$black,
  opacity: 0.8,
});

export const APVoteImgBkgndCoverGradient = glamorous.div({
  label: 'AP-vote-img-bkgnd-cover-gradient',
  position: 'absolute',
  left: 0,
  bottom: 0,
  width: '100%',
  height: '60%',
  background: `linear-gradient(to bottom, rgba(255,255,255,0) 0%,rgba(0,0,0,1) 100%)`,
});

export const APVoteHeadline = glamorous(APHeadline)({
  label: 'AP-vote-headline',
  textTransform: 'none',
  textAlign: 'left',
  fontSize: 38,
});

export const APVoteContent = glamorous.div({
  label: 'AP-vote-content',
  width: '100%',
  height: '100vh',
  position: 'absolute',
  top: 0,
  left: 0,
  padding: '90px 20px 0',
});

export const APVoteSubHeadline = glamorous(APVoteHeadline)({
  label: 'AP-vote-sub-headline',
  [mediaQueries.mobileInner]: {
    textAlign: 'center',
    fontSize: 40,
    padding: '25px 0',
  },
});

// Option bar

export const APOptionBar = glamorous.div({
  label: 'AP-option-bar',
  position: 'relative',
  marginBottom: 12,
  display: 'flex',
});

export const APOptionBarLabel = glamorous.div<{ bkgnd: string }>(
  {
    label: 'AP-option-bar-label',
    display: 'flex',
    alignItems: 'center',
    height: 44,
    width: '100%',
    paddingLeft: 8,
  },
  ({ bkgnd }) => ({
    background: bkgnd,
  }),
);

export const APOptionBarIcon = glamorous.img({
  label: 'AP-option-bar-icon',
  width: 35,
  display: 'inline-block',
  marginRight: 20,
});

export const APOptionBarIconLabel = glamorous(P)({
  label: 'AP-option-bar-icon-label',
  display: 'inline-block',
  color: colours.$white,
  marginBottom: 0,
});

export const APOptionBarCta = glamorous.div<{ bkgnd: string }>(
  {
    label: 'AP-option-bar-cta',
    display: 'inline-block',
    width: 52,
    height: 44,
    marginLeft: 5,
  },
  ({ bkgnd }) => ({
    background: bkgnd,
  }),
);

export const APOptionBarCtaIcon = glamorous.span({
  label: 'AP-option-bar-cta-icon',
  display: 'inline-block',
  position: 'relative',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
});

// Results Screen

export const APResultsHeadline = glamorous(APVoteHeadline)({
  label: 'AP-results-headline',
  textTransform: 'none',
  textAlign: 'center',
  fontSize: 38,
  [mediaQueries.mobileInner]: {
    fontSize: 40,
    padding: '25px 0',
  },
});

export const APChartHolder = glamorous.div({
  label: 'AP-option-bar-cta-icon',
  width: '100%',
  height: 330,
  marginBottom: 10,
});

// Progress bar

export const APProgressBar = glamorous.div({
  label: 'AP-progress-bar',
  width: '100%',
  display: 'flex',
  justifyContent: 'space-around',
  padding: '0 13px',
  [mediaQueries.tabletPortrait]: {
    padding: '0 25px',
  },
});

export const APProgressBarItem = glamorous.div<{ viewed: boolean }>(
  {
    label: 'AP-progress-bar-item',
    background: colours.$white,
    height: 4,
    margin: '0 3px',
    flexGrow: 1,
    [mediaQueries.tabletPortrait]: {
      height: 6,
    },
  },
  ({ viewed }) => ({
    opacity: viewed ? 1.0 : 0.5,
  }),
);

// Navigation

export const APNav = glamorous.div({
  label: 'AP-nav',
  width: '100%',
  display: 'flex',
  justifyContent: 'space-between',
});

export const APNavIcon = glamorous.div<{ disabled: boolean }>(
  {
    label: 'AP-nav-icon',
    border: `3px solid ${colours.$white}`,
    width: 35,
    height: 35,
    borderRadius: '50%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  ({ disabled }) => ({
    opacity: disabled ? 0.5 : 1.0,
  }),
);

// Poll Container

export const APNavHolder = glamorous.div({
  label: 'AP-nav-holder',
  width: 'calc(100% - 32px)',
  position: 'absolute',
  bottom: 110,
  left: 16,
  // right: 16,
  zIndex: 5,
  [mediaQueries.tabletPortrait]: {
    width: '100%',
    left: 0,
    bottom: 16,
    right: 'auto',
    padding: '0 12px',
  },
});

export const APProgressHolder = glamorous.div({
  label: 'AP-progress-holder',
  width: '100%',
  position: 'absolute',
  top: 8,
  left: 0,
  zIndex: 5,
  [mediaQueries.tabletPortrait]: {
    top: 12,
  },
});

export const APNavIntroHolder = glamorous(APNavHolder)({
  label: 'AP-nav-intro-holder',
  width: 'auto',
  right: 16,
  left: 'auto',
  bottom: 185,
  [mediaQueries.tabletPortrait]: {
    left: 'auto',
    right: 0,
    width: 105,
    display: 'flex',
    justifyContent: 'flex-end',
  },
});
