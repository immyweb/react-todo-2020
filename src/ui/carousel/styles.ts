import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
