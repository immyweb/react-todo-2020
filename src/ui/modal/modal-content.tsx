import * as React from 'react';

import { P } from '../typography/typography';
import { ModalHeadline } from '../modal/styles';
import { Image } from '../../styles/globals';
import { IModalContent } from '../../types/modal-content';

export const ModalContent = (modalContent: IModalContent) => {
  const { title, imageTop, copy, imageBottom } = modalContent;
  return (
    <div className="modal-content">
      <ModalHeadline>{title}</ModalHeadline>
      {imageTop && <Image src={imageTop} alt={title} marginBottom={10} />}
      {copy &&
        copy.map((paragraph: string, i: number) => {
          return (
            <P txtalign={'center'} key={i}>
              {paragraph}
            </P>
          );
        })}
      {imageBottom && <Image src={imageBottom} alt={title} marginBottom={10} />}
    </div>
  );
};
