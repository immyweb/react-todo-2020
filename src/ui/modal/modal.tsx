import * as React from 'react';
import Measure from 'react-measure';

import { ModalContainer, ModalInner, ModalContent, BtnClose, BtnOpen } from './styles';
import { setSwipeEnabled } from '../../utils/modal-fix';

import { ReactComponent as CloseIcon } from '../images/icons/close-solid.svg';

interface IProps {
  content: React.ReactNode;
  standalone: boolean;
  openTxt?: string;
  active?: boolean;
  onCloseModal?: () => void;
}

interface IState {
  showModal: boolean;
  outerHeight: number;
  innerHeight: number;
  innerTaller: boolean;
}

export class Modal extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      showModal: props.active ? props.active : false,
      outerHeight: 0,
      innerHeight: 0,
      innerTaller: false,
    };
  }

  UNSAFE_componentWillMount() {
    setSwipeEnabled(true);
  }

  componentDidMount() {
    this.setState({ outerHeight: window.innerHeight - 40 }); // subtrack padding top/bottom
  }

  componentWillUnmount() {
    this.setState({
      showModal: false,
      innerTaller: false,
      outerHeight: 0,
    });

    setSwipeEnabled(false);
  }

  onClose() {
    if (this.props.onCloseModal) {
      this.props.onCloseModal();
    }
  }

  toggle() {
    this.setState({
      showModal: !this.state.showModal,
    });
  }

  render() {
    const { standalone, openTxt, content } = this.props;
    return (
      <Measure
        bounds
        onResize={(contentRect) => {
          const iHeight = contentRect.bounds!.height;
          this.setState({ innerHeight: iHeight + 110 }); // add padding top/bottom
          if (this.state.innerHeight > this.state.outerHeight) {
            this.setState({ innerTaller: true });
          }
        }}
      >
        {({ measureRef }) => (
          <div>
            {standalone && <BtnOpen onClick={() => this.toggle()}>{openTxt ? openTxt : 'Open'}</BtnOpen>}
            <ModalContainer open={this.state.showModal}>
              <ModalInner innerTaller={this.state.innerTaller}>
                <ModalContent innerTaller={this.state.innerTaller}>
                  <div ref={measureRef}>{content}</div>
                </ModalContent>
                <BtnClose onClick={standalone ? () => this.toggle() : () => this.onClose()}>
                  <CloseIcon />
                </BtnClose>
              </ModalInner>
            </ModalContainer>
          </div>
        )}
      </Measure>
    );
  }
}
