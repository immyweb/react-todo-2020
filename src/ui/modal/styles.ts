import glamorous from 'glamorous';

import Button from '../buttons/buttons';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { fonts } from '../../styles/fonts';

import { rgba, transparentize } from 'polished';

interface IGProps {
  open?: boolean;
  height?: string;
  innerTaller?: boolean;
}

export const ModalContainer = glamorous.div<IGProps>(
  {
    label: 'modal-container',
    position: 'fixed',
    width: '100%',
    height: '100%',
    backgroundColor: rgba(colours.$black, 0.5),
    top: 0,
    left: 0,
    zIndex: 160,
  },
  ({ open }) => ({
    visibility: open ? 'visible' : 'hidden',
  }),
);

export const ModalInner = glamorous.div<IGProps>(
  {
    label: 'modal-inner',
    position: 'relative',
    height: '95%',
    width: '90%',
    top: '50%',
    transform: 'translateY(-50%)',
    background: colours.$white,
    margin: '0 auto',
    padding: '20px 20px 30px 20px',
    overflowY: 'hidden',
    msOverflowStyle: '-ms-autohiding-scrollbar',
    borderRadius: 2,
    [mediaQueries.tabletPortrait]: {
      width: 620,
    },
  },
  ({ innerTaller }) => ({
    height: innerTaller ? '95%' : 'auto',
    [mediaQueries.tabletPortrait]: {
      height: innerTaller ? '80%' : 'auto',
    },
  }),
);

export const ModalContent = glamorous.div<IGProps>(
  {
    label: 'modal-content',
    position: 'relative',
    overflowY: 'auto',
    msOverflowStyle: '-ms-autohiding-scrollbar',
    height: '100%',
    ':after': {
      content: ' ',
      width: '100%',
      position: 'fixed',
      left: 0,
      bottom: 28,
      background: `linear-gradient(to top, ${transparentize(0, colours.$white)} 20%, ${transparentize(
        1,
        colours.$white,
      )})`,
      zIndex: 161,
    },
  },
  ({ innerTaller }) => ({
    paddingBottom: innerTaller ? 60 : 0,
    ':after': {
      height: innerTaller ? 60 : 0,
    },
  }),
);

export const ModalHeadline = glamorous.h2({
  label: 'modal-headline',
  textTransform: 'uppercase',
  textAlign: 'center',
  fontSize: 24,
  fontFamily: fonts.$theSunHeavyCondensed,
  marginBottom: 20,
  lineHeight: 1.1,
  padding: '0 25px',
  [mediaQueries.tabletPortrait]: {
    padding: 0,
  },
});

export const BtnClose = glamorous.div({
  label: 'modal-btn-close',
  textAlign: 'center',
  position: 'absolute',
  top: 10,
  right: 10,
  height: 28,
  width: 28,
  background: colours.$white,
  borderRadius: '50%',
  ':hover': {
    cursor: 'pointer',
  },
});

export const BtnOpen = glamorous(Button)({
  label: 'modal-btn-open',
});
