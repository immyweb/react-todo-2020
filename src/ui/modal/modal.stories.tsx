import * as React from 'react';

import { storiesOf } from '@storybook/react';

import { Modal } from './modal';
import { H2, P } from '../typography/typography';

const ModalContent = () => {
  return (
    <div className="modal-content">
      <H2>SNOW IN THE DESERT</H2>
      <P>Explain why the person is in this spot as it is obviously words placed distion.</P>
      <img src="http://via.placeholder.com/580x250" alt="" />
      <P>
        Kirk Pasich, a lawyer for the family, said: “Without the results of toxicology tests, we do
        not know what was going on with Chris—or if any to his demise.
      </P>
      <P>
        “The family believes that if Chris took his life, he did not know what he was doing, and
        that drugs or other was doing, that drugs or other substances may have affected his
        actions.”
      </P>
      <P>
        “Without the results of toxicology tests, we do not know what was going on with Chris—or if
        any substances contributed to his demise.
      </P>
      <P>
        Kirk Pasich, a lawyer for the family, said: “Without the results of toxicology tests, we do
        not know what was going on with Chris—or if any to his demise.
      </P>
    </div>
  );
};

storiesOf('Modal', module).add('Modal', () => (
  <Modal content={<ModalContent />} standalone={true} active={false} />
));
