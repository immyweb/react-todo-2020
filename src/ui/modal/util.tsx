import * as React from 'react';
import * as DOM from 'react-dom';
import { podTracker } from '../../utils/pod-tracker-utils';

import { Modal } from '../modal/modal';

import { IModalContent } from '../../types/modal-content';

export const onCloseModal = (domId: string, title: string, id?: string) => {
  podTracker.click({
    id,
    description: `hotspot modal "${title}" closed`,
  });
  DOM.unmountComponentAtNode(document.getElementById(domId) as any);
};

export const onOpenModal = (
  modalData: IModalContent,
  domId: string,
  ModalContentRef: any,
  id?: string,
) => {
  if (modalData) {
    const content = ModalContentRef(modalData);
    podTracker.click({
      id,
      description: `hotspot modal "${modalData.title}" opened`,
    });
    DOM.render(
      <Modal
        content={content}
        standalone={false}
        active={true}
        onCloseModal={() => onCloseModal(domId, modalData.title, id)}
      />,
      document.getElementById(domId),
    );
  }
};
