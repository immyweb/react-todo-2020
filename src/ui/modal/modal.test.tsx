import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { Modal } from './modal';
import { BtnOpen, BtnClose } from './styles';

const ModalContent = () => {
  return (
    <div className="modal-content">
      <h2>SNOW IN THE DESERT</h2>
      <h3>Explain why the person is in this spot as it is obviously words placed distion.</h3>
      <img src="http://via.placeholder.com/580x312" alt="" />
      <p>
        Kirk Pasich, a lawyer for the family, said: “Without the results of toxicology tests, we do
        not know what was going on with Chris—or if any to his demise.
      </p>
      <p>
        “The family believes that if Chris took his life, he did not know what he was doing, and
        that drugs or other was doing, that drugs or other substances may have affected his
        actions.”
      </p>
      <p>
        “Without the results of toxicology tests, we do not know what was going on with Chris—or if
        any substances contributed to his demise.
      </p>
      <p>
        Kirk Pasich, a lawyer for the family, said: “Without the results of toxicology tests, we do
        not know what was going on with Chris—or if any to his demise.
      </p>
    </div>
  );
};

function shallowModal(Content: any, openTxt?: string) {
  return shallow(<Modal content={Content} openTxt={openTxt} standalone={true} />);
}

function mountModal(Content: any, openTxt?: string) {
  return mount(<Modal content={Content} openTxt={openTxt} standalone={true} />);
}

describe('<Modal />', () => {
  it('displays passed in content', () => {
    const wrapper = mountModal(<ModalContent />);
    const innerContent = wrapper.find(ModalContent);
    expect(innerContent.length).toBe(1);
  });

  it('clicking open button triggers modal to open', () => {
    const wrapper = mountModal(<ModalContent />);
    const openBtn = wrapper.find(BtnOpen);
    expect(wrapper.state('showModal')).toBe(false);
    openBtn.simulate('click');
    expect(wrapper.state('showModal')).toBe(true);
  });

  it('clicking on close button closes modal', () => {
    const wrapper = mountModal(<ModalContent />);
    const openBtn = wrapper.find(BtnOpen);
    openBtn.simulate('click');
    expect(wrapper.state('showModal')).toBe(true);
    const closeBtn = wrapper.find(BtnClose);
    closeBtn.simulate('click');
    expect(wrapper.state('showModal')).toBe(false);
  });

  it('if no button text is provided, it should display default text', () => {
    const wrapper = shallowModal(<ModalContent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('if button text is provided, it should display the provided text', () => {
    const wrapper = shallowModal(<ModalContent />, 'Click me');
    expect(wrapper).toMatchSnapshot();
  });

  it('toogle function is called when open button is clicked', () => {
    const wrapper = mountModal(<ModalContent />);
    const toggleFn = jest.spyOn(wrapper.instance() as Modal, 'toggle');

    const openBtn = wrapper.find(BtnOpen);
    openBtn.simulate('click');
    expect(toggleFn).toHaveBeenCalledTimes(1);
  });

  it('toogle function is called when close button is clicked', () => {
    const wrapper = mountModal(<ModalContent />);
    const toggleFn = jest.spyOn(wrapper.instance() as Modal, 'toggle');

    const closeBtn = wrapper.find(BtnClose);
    closeBtn.simulate('click');
    expect(toggleFn).toHaveBeenCalledTimes(1);
  });
});
