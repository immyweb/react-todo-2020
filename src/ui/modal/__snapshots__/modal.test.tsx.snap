// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> if button text is provided, it should display the provided text 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    content={<ModalContent />}
    openTxt="Click me"
    standalone={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bounds": true,
      "children": [Function],
      "onResize": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bounds": true,
        "children": [Function],
        "onResize": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Modal /> if no button text is provided, it should display default text 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    content={<ModalContent />}
    standalone={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bounds": true,
      "children": [Function],
      "onResize": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bounds": true,
        "children": [Function],
        "onResize": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
